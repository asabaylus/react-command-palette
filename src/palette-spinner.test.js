/*  eslint
  no-unused-vars: ["error", { "varsIgnorePattern": "^renderer$" }],
  "function-paren-newline":0  */

import * as React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import serializer from "enzyme-to-json/serializer";
import PaletteSpinner from "./palette-spinner";

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(serializer);

describe("PaletteSpinner", () => {
  describe("props.spinner", () => {
    it("should render the DefaultSpinnerComponent by default", () => {
      const wrapper = mount(<PaletteSpinner />);
      const spinner = wrapper.find("DefaultSpinnerComponent");
      expect(spinner).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it("should display a custom string when props.spinner is set", () => {
      const wrapper = mount(<PaletteSpinner spinner="Waiting" />);
      const spinner = wrapper.find("CustomSpinnerComponent");
      expect(spinner).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it('should display an "inline" custom string when props.spinner is set', () => {
      const wrapper = mount(
        <PaletteSpinner spinner="Waiting" display="inline" />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("props.display", () => {
    it("should display a black spinner when set to inline", () => {
      const wrapper = mount(<PaletteSpinner display="inline" />);
      const hasCSSInlineClass = wrapper.find("div").hasClass("inline");
      expect(hasCSSInlineClass).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it("should display a white spinner when set to modal", () => {
      const wrapper = mount(<PaletteSpinner display="modal" />);
      const hasCSSInlineClass = wrapper.find("div").hasClass("modal");
      expect(hasCSSInlineClass).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
