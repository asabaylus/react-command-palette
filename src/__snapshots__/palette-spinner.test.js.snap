// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteSpinner props.display should display a black spinner when set to inline 1`] = `
<PaletteSpinner
  display="inline"
>
  <DefaultSpinnerComponent
    display="inline"
  >
    <div
      className="default-spinner inline"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </DefaultSpinnerComponent>
</PaletteSpinner>
`;

exports[`PaletteSpinner props.display should display a white spinner when set to modal 1`] = `
<PaletteSpinner
  display="modal"
>
  <DefaultSpinnerComponent
    display="modal"
  >
    <div
      className="default-spinner modal"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </DefaultSpinnerComponent>
</PaletteSpinner>
`;

exports[`PaletteSpinner props.spinner should display a custom string when props.spinner is set 1`] = `
<PaletteSpinner
  spinner="Waiting"
>
  <CustomSpinnerComponent
    spinner="Waiting"
  >
    <div
      className="spinner modal"
      role="status"
    >
      Waiting
    </div>
  </CustomSpinnerComponent>
</PaletteSpinner>
`;

exports[`PaletteSpinner props.spinner should display an "inline" custom string when props.spinner is set 1`] = `
<PaletteSpinner
  display="inline"
  spinner="Waiting"
>
  <CustomSpinnerComponent
    display="inline"
    spinner="Waiting"
  >
    <div
      className="spinner inline"
      role="status"
    >
      Waiting
    </div>
  </CustomSpinnerComponent>
</PaletteSpinner>
`;

exports[`PaletteSpinner props.spinner should render the DefaultSpinnerComponent by default 1`] = `
<PaletteSpinner>
  <DefaultSpinnerComponent>
    <div
      className="default-spinner modal"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </DefaultSpinnerComponent>
</PaletteSpinner>
`;
